discovery.md

```diff
@@ -473,13 +515,14 @@ When encoded in JSON, the response format MUST adhere to the following:
 
 ```
 {
-  "url": "SERVICE-URL",
+  "id": "{id}",
+  "url": "{url}",
   "name": "{name}",
   ... remainder of Service attributes ...
 }
 ```
 
-##### `/services?matching=[search term]`
+##### `/services?name=[search term]`
```

Comment:
Two things:

1. I think ?name= and ?matching= is not the same
2. the "contains"-style query does not functionally replace /service/{name} (from above) because here you'd get all services whose name contains the search term

I'd suggest to keep ?name= here but change the description. REST query parameters like "?name=", to me personally, provide only exact matches anyhow, so I would no longer expect it to return all Services that "contain" the search term.
But using ?matching, IMHO is also not a good encoding of a search operation operation. Personally, I don't like encoding the operation type (match) in a URL query parameter ("?matching") and at the same time imply the searched field (name). Swagger suggests to encode operations as endpoints, e.g. /services/matching.

So, with ?name= it might be better to change the functionality here to offer an **exact** match in order to provide an equivalent to /services/{name}.
But then, a replacement for the ?matching= functionality is also required to return "contains" matches. Now that /services/{name} is gone, it could be /services/matchName, /services/match?name=, /services/find?name= or any /services/{operation} as long as it's not a UUID.

Reply:
I agree with your logic on "name" - defining it to mean "matching' is a bit awkward. 
I'm not sure yet how I feel about `/service/{operation}` - I think that might require a bit more thinking by the group. In the interest of making forward progress though, I'd like to propose that as part of this PR we use "name" to mean an exact match - then we still allow for exact-name-based searching. Then I'll open an issue for the "matching" stuff so we can brainstorm on how people would like to add that.
What do people think?